src/main/java/org/jitsi/jicofo/recording/jibri/JibriSession.java

```diff
@@ -443,24 +440,62 @@ private void reschedulePendingTimeout()
     }
 
     /**
-     * Stores current Jibri status and notifies {@link #owner}.
-     * @param newStatus the new Jibri status to be set
-     * @param failureReason optional error for failed state.
+     * Handle a Jibri status update (this could come from an IQ response, a new IQ from Jibri, an XMPP event, etc.).
+     * This will handle:
+     * 1) Retrying with a new Jibri in case of an error
```

Comment:
can you explain how will Jicofo try another Jibri exactly ? how the `jibriDetector` will know to try another one ?

Reply:
do you mean just elaborate in the comment?  we just call `start` here which contains the logic to find an available jibri

Reply:
I mean it's very likely that the selector will return the same one unless it would report a failure immediately

Reply:
or do we rely on that it would be in the busy state ?

Reply:
from jicofo's side we assume that a jibri that has caused a failed session will have marked itself as no longer available for use.  from jibri, we bubble the error up (the same one that resulted in the failure here) to our status and mark ourselves as unhealthy.  (note that this is new behavior for jibri and it has just started doing this as of https://github.com/jitsi/jibri/pull/159)